Remove duplicate CSS from Navigation that is aleady in Navigation Link CSS

jeryj opened the pull request:
Closes #19537

There was a bit of repeated code in navigation/editor.scss already appearing in navigation-link/editor.scss. The specificity will end up being a little lower now, but from my testing I don't see anything that overrides it.

## Description
I've removed CSS from the Navigation Block that already appeared in Navigation Link CSS.

## How has this been tested?
Manually

## Types of changes
Minor refactor of CSS.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->

jeryj committed:
Deleted CSS related to .wp-block-navigation-link in navigation/editor.scss already appearing in navigation-link/editor.scss

There was a bit of repeated code in navigation/editor.scss already appearing in navigation-link/editor.scss. The specificity will end up being a little lower now, but from my testing I don't see anything that overrides it.

jeryj commented:
The last item is: Can we move the remaining CSS related to `.wp-block-navigation-link` in `packages/block-library/src/navigation/editor.scss` to `packages/block-library/src/navigation-link/editor.scss`?

The only reason I can see that this would be an issue is if `.wp-block-navigation-link` is ever used without a parent wrapper of `.wp-block-navigation`.

The remaining CSS to maybe be moved over from `navigation/editor.scss` is:

```
.wp-block-navigation .wp-block-navigation-link {
	margin-left: $grid-size;

	// Provide a base navigation item margin.
	// This should be the same inside the field,
	// and the edit container should compensate.
	// This is to make sure the edit and view are the same.
	padding: 0 $grid-size;
}
```

But some of this is even repeated already in `/navigation-link/editor.scss`:
```
.wp-block-navigation-link {
	margin-right: $grid-size;
	// Provide a base menu item margin.
	// This should be the same inside the field,
	// and the edit container should compensate.
	// This is to make sure the edit and view are the same.
	padding: 0 $grid-size;
	padding-right: 55px; // allow space for Block "mover"
```

My thought is that if `.wp-block-navigation-link` is never used outside of `.wp-block-navigation`, we can:
- remove the `padding-right: 55px; // allow space for Block "mover"` line 
- add `margin-left: $grid-size;`
- remove the remaining `.wp-block-navigation .wp-block-navigation-link` CSS from `navigation/editor.scss`

obenland commented:
I think navigation link always being inside navigation is a safe assumption for now. If that ever changes, any necessary adjustments can be made then

jeryj committed:
Moved remaining .wp-block-navigation-link CSS from navigation to navigation-link

There was a little remaining CSS in navigation/editor.scss that I moved over to navigation-link/editor.scss to keep it all in one place. I also removed the extra right padding on the .wp=block-navigation-link CSS that is never applied due to specificity.

jeryj commented:
@obenland, Thanks for the follow-up! I've moved the remaining `.wp-block-navigation-link` CSS into `navigation-link/editor.scss`.

obenland reviewed (approved):
Nice work!
